packages/math/base-elements.lua

```diff
@@ -1334,6 +1334,79 @@ end
 
 function elements.table.output (_) end
 
+elements.sqrt = pl.class(elements.mbox)
+elements.sqrt._type = "Sqrt"
+
+function elements.sqrt:__tostring ()
+   return self._type .. "(" .. tostring(self.radicand) .. ")"
+end
+
+function elements.sqrt:_init (radicand)
+   elements.mbox._init(self)
+   self.radicand = radicand
+   table.insert(self.children, radicand)
+   self.relX = SILE.types.length(0) -- x position relative to its parent box
+   self.relY = SILE.types.length(0) -- y position relative to its parent box
+end
+
+function elements.sqrt:styleChildren ()
+   self.radicand.mode = self.mode
+end
+
+function elements.sqrt:shape ()
+   local mathMetrics = self:getMathMetrics()
+   local scaleDown = self:getScaleDown()
+   local constants = mathMetrics.constants
+
+   self.radicalRuleThickness = constants.radicalRuleThickness * scaleDown
+   if self.mode == mathMode.display or self.mode == mathMode.displayCramped then
+      self.radicalVerticalGap = constants.radicalDisplayStyleVerticalGap * scaleDown
+   else
+      self.radicalVerticalGap = constants.radicalVerticalGap * scaleDown
+   end
+   self.extraAscender = constants.radicalExtraAscender * scaleDown
+
+   -- HACK: More or less ad hoc values, see output method for more details
+   self.symbolWidth = SILE.shaper:measureChar("√").width
+   self.symbolHeight = SILE.types.length("1.2ex"):tonumber() * scaleDown
+
+   self.width = self.radicand.width + self.symbolWidth
+   self.height = self.radicand.height + self.radicalVerticalGap + self.extraAscender
+   self.depth = self.radicand.depth
+   self.radicand:shape()
+   self.radicand.relX = self.symbolWidth
+end
+
+function elements.sqrt:output (x, y, line)
+   -- HACK FIXME:
+   -- OpenType might say we need to assemble the radical sign from parts.
+   -- Frankly, it's much easier to just draw it as a graphic :-)
+   -- Hence, here we use a PDF graphic operators to draw a nice radical sign.
+   -- Some values here are ad hoc, but they look good.
+   local h = self.height:tonumber()
+   local d = self.depth:tonumber()
+   local sw = self.symbolWidth
+   local dh = h - self.symbolHeight
+   local symbol = {
+      self.radicalRuleThickness, "w", -- line width
+      2, "j", -- round line joins
+      sw, self.extraAscender, "m",
+      sw * 0.4, h + d, "l",
+      sw * 0.15, dh,  "l",
+      0, dh + 0.5, "l",
```

Comment:
We actually need a rounding helper here, some PDF readers do not like double precision numbers at all.

```
local function _r (number)
  -- Lua 5.3+ formats floats as 1.0 and integers as 1
  -- Also some PDF readers do not like double precision.
  return math.floor(number) == number and math.floor(number) or tonumber(string.format("%.5f", number))
end
```

Reply:
Fixed!
I'm done here with this (inventive but) poor man's square root ;)

Reply:
I see code in `elements.text:stretchyReshape` that uses math variants, so why not use it to draw the radical glyph?

Reply:
> ... so why not use it to draw the radical glyph?

'Cause it didn't happen in 2+ years? It seems to me something quick is better than nothing, then...

Maybe too 'cause the stretchy reshape has a big TODO in its code: it doesn't do the glyph assembling where there's no variant available (e.g. no big enough delimiter), and this is going to be another hornets' nest on its own.